src/conformist.ml

```diff
@@ -193,83 +238,83 @@ let rec fold_left'
 
 let fold_left ~f ~init schema = fold_left' ~f ~init schema.fields
 
-type error = string * string option * string
+type error = string * string list * string
 type input = (string * string list) list
 
-let validate schema input =
+let validate
+    (schema : ('meta, 'ctor, 'ty) t)
+    (input : (string * string list) list)
+    : error list
+  =
   let f errors field =
     let name = Field.name field in
     match List.assoc name input with
-    | [ value_string ] ->
-      (match Field.validate field value_string with
-      | Some msg -> List.cons (name, Some value_string, msg) errors
-      | None -> errors)
     | values ->
-      let value = Format.sprintf "[%s]" (String.concat ", " values) in
-      List.cons (name, Some value, "Multiple values provided") errors
+      (match Field.validate field values with
+      | Some msg -> List.cons (name, values, msg) errors
+      | None -> errors)
     | exception Not_found ->
```

Comment:
I didn't know you could match on exceptions :o What's the point of `try` then?

Reply:
I think `try` is a shorthand and `match` is the general case